src/core/__tests__/nested-option.test.tsx

```diff
@@ -19,124 +29,287 @@ class CollectionNestedComponent extends ConfigurationComponent<{ c?: number, d?:
 }
 // tslint:enable:max-classes-per-file
 
-it("pulls options from a single nested component", () => {
-    mount(
-        <TestComponent>
-            <NestedComponent1 a={123} />
-        </TestComponent>
-    );
-
-    expect(WidgetClass.mock.calls[0][1]).toEqual({
-        templatesRenderAsynchronously: true,
-        option1: {
-            a: 123
-        }
+describe("nested option", () => {
+
+    it("is pulled", () => {
+        mount(
+            <TestComponent>
+                <NestedComponent a={123} />
+            </TestComponent>
+        );
+
+        expect(WidgetClass.mock.calls[0][1]).toEqual({
+            templatesRenderAsynchronously: true,
+            option: {
+                a: 123
+            }
+        });
     });
-});
 
-it("doesn't pull options from wrong component", () => {
-    mount(
-        <TestComponent>
-            <NestedComponent1 a={123} />
-            <WrongNestedComponent x={456} />
-        </TestComponent>
-    );
-
-    expect(WidgetClass.mock.calls[0][1]).toEqual({
-        templatesRenderAsynchronously: true,
-        option1: {
-            a: 123
-        }
+    it("is pulled (several options)", () => {
+        mount(
+            <TestComponent>
+                <NestedComponent a={123} />
+                <WrongNestedComponent x={456} />
+                <AnotherNestedComponent b="abc" />
+            </TestComponent>
+        );
+
+        expect(WidgetClass.mock.calls[0][1]).toEqual({
+            templatesRenderAsynchronously: true,
+            option: {
+                a: 123
+            },
+            anotherOption: {
+                b: "abc"
+            }
+        });
     });
-});
 
-it("pulls overriden options from the same nested component", () => {
-    mount(
-        <TestComponent>
-            <NestedComponent1 a={123} />
-            <NestedComponent1 a={456} />
-        </TestComponent>
-    );
-
-    expect(WidgetClass.mock.calls[0][1]).toEqual({
-        templatesRenderAsynchronously: true,
-        option1: {
-            a: 456
-        }
+    it("isn't pulled by wrong parent", () => {
+        mount(
+            <TestComponent>
+                <NestedComponent a={123} />
+                <WrongNestedComponent x={456} />
+            </TestComponent>
+        );
+
+        expect(WidgetClass.mock.calls[0][1]).toEqual({
+            templatesRenderAsynchronously: true,
+            option: {
+                a: 123
+            }
+        });
     });
-});
 
-it("pulls array options from a nested component", () => {
-    mount(
-        <TestComponent>
-            <CollectionNestedComponent c={123} d="abc" />
-        </TestComponent>
-    );
-
-    expect(WidgetClass.mock.calls[0][1]).toEqual({
-        templatesRenderAsynchronously: true,
-        itemOptions: [
-            { c: 123, d: "abc" }
-        ]
+    it("is pulled overriden if not a collection item", () => {
+        mount(
+            <TestComponent>
+                <NestedComponent a={123} />
```

Comment:
Maybe, in this case, we should also throw an exception.

Reply:
Added this to our inbox.